### PR TITLE
Overlay component

### PR DESCRIPTION
**Pull request by deadlybutter**
This pull request adds an Overlay component. This component will present Notifications, Modals and anything else we want floating above the screen.

### CODE REVIEW

File: resources/assets/init.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ import { observerMiddleware } from './middleware/analytics';
 const history = historyInit();
 const middleware = [thunk, routerMiddleware(history), observerMiddleware];
 const store = configureStore({ ...reducers, routing: routerReducer }, middleware, window.STATE);
-
+store.dispatch({ type: 'ADD_NOTIFICATION' });
```
Comment (deadlybutter) [inline resources/assets/init.js]: delete this!

**Comment by deadlybutter** [comment]
(Also: Take note at how i named the new component files, let me know whatcha think)

**Comment by weerd** [comment]
mind popping up an example screenshot? Helps w/ visualizing some of the css code :)

**Comment by weerd** [comment]
🤦‍♂️ did it again

**Comment by deadlybutter** [comment]
@weerd you're a monster, and yeah sure thing, one sec

**Comment by deadlybutter** [comment]
@weerd So the first screenshot is the overlay itself,

screen shot 2017-06-06 at 10 12 49 am

And this one is the individual components within the overlay

screen shot 2017-06-06 at 10 13 09 am

**Comment by weerd** [comment]
So, there can be multiple items inside the overlay right? Does each of them have their own close `X` button? If so, then is there an additional close `X` button for the overlay; otherwise how do you close it?

**Comment by deadlybutter** [comment]
Yeah, the close functionality is handled outside of the overlay. Modals for example will behave differently than the notifications, because clicking the black background is supposed to close it.

**Comment by weerd** [comment]
Does it just auto-close when all things inside it have been closed?

**Comment by deadlybutter** [comment]
@weerd There technically is no "close" here, if there are no children it just doesn't render anything

**Pull request merged**